dvc/ignore.py

```diff
@@ -85,20 +108,46 @@ def ignore(self, path, is_dir):
                     result = ignore
         return result
 
+    def match_details(self, dirname, basename, is_dir=False):
+        path = self._get_normalize_path(dirname, basename)
+        return self._ignore_details(path, is_dir)
+
+    def _ignore_details(self, path, is_dir):
+        result = []
+        for ignore, pattern in zip(self.regex_pattern_list, self.pattern_list):
+            regex = re.compile(ignore[0])
+            # skip system pattern
+            if not pattern.file_info:
+                continue
```

Comment:
Won't that result in no matches for `.git` and return 1 when we do `check-ignore .git`?
I guess in one way this is right behavior, since its not specified in any file, but, its also against `tree.walk` behavior which will ignore `.git`. 
But I guess main point of `check-ignore` is to debug `.dvcignore`s, so its fine.

Reply:
I just followed Git.
![image](https://user-images.githubusercontent.com/6745454/88608014-56128880-d0b3-11ea-9384-2700f3d192da.png)
Maybe I can return 
```
$ dvc check-ignore .git
::System pattern   .git
$ echo $?
0
```

Reply:
If that's not too much trouble, sure. But if its, feel free to keep it that way.

Reply:
![image](https://user-images.githubusercontent.com/6745454/88624966-83266180-d0da-11ea-8ab9-523074c1284b.png)
I think this is the same problem with the above one.
If 0 or 1 depends on whether the file is excluded, then we should return 0 here, but If it depends on whether any pattern in .dvcignore matches the file, then we should return 1 here.

Reply:
@karajan1001 you mean the proper git return code?
To avoid unnecessary change requests, let's leave it as is.